libraries/shared/watcher/event_watcher_test.go

```diff
@@ -208,5 +208,15 @@ var _ = Describe("Event Watcher", func() {
 
 			Expect(err).To(MatchError(fakes.FakeError))
 		})
+
+		It("doesn't panic if one of the go routines errors and closes the err channel", func() {
+			extractor.ExtractLogsErrors = []error{nil, errExecuteClosed}
+			delegator.DelegateErrors = []error{nil,  errExecuteClosed, errExecuteClosed}
+
+			err := eventWatcher.Execute(constants.HeaderUnchecked)
+
+			Expect(err).To(MatchError(errExecuteClosed))
+			Expect(delegator.DelegateCallCount > 0).To(BeTrue())
```

Comment:
I don't know if this is realistic but it seems maybe(?) possible that the extract call could trigger things to quit before delegate is called. Wondering if it might be worth asserting that _either_ extract call count _or_ delegate call count is > 0